modules/RTC/TraceablePeerConnection.js

```diff
@@ -1123,117 +1130,80 @@ TraceablePeerConnection.prototype._removeRemoteTrack = function(toBeRemoved) {
 };
 
 /**
- * Returns a map with keys msid/mediaType and <tt>TrackSSRCInfo</tt> values.
- * @param {RTCSessionDescription} desc the local description.
- * @return {Map<string,TrackSSRCInfo>}
+ * Processes the local SDP and creates an SSRC map for every local track.
+ *
+ * @param {string} localSDP - SDP from the local description.
+ * @returns {void}
  */
-TraceablePeerConnection.prototype._extractSSRCMap = function(desc) {
-    /**
-     * Track SSRC infos mapped by stream ID (msid) or mediaType (unified-plan)
-     * @type {Map<string,TrackSSRCInfo>}
-     */
+TraceablePeerConnection.prototype._processAndExtractSourceInfo = function(localSDP) {
     const ssrcMap = new Map();
 
-    /**
-     * Groups mapped by primary SSRC number
-     * @type {Map<number,Array<SSRCGroupInfo>>}
-     */
-    const groupsMap = new Map();
-
-    if (typeof desc !== 'object' || desc === null
-        || typeof desc.sdp !== 'string') {
-        logger.warn('An empty description was passed as an argument');
-
+    if (!localSDP || typeof localSDP !== 'string') {
         return ssrcMap;
     }
-
-    const session = transform.parse(desc.sdp);
-
-    if (!Array.isArray(session.media)) {
-        return ssrcMap;
-    }
-
-    let media = session.media;
-
-    media = media.filter(mline => mline.direction === MediaDirection.SENDONLY
+    const session = transform.parse(localSDP);
+    const media = session.media.filter(mline => mline.direction === MediaDirection.SENDONLY
         || mline.direction === MediaDirection.SENDRECV);
 
-    let index = 0;
-
-    for (const mLine of media) {
-        if (!Array.isArray(mLine.ssrcs)) {
-            continue; // eslint-disable-line no-continue
-        }
+    if (!Array.isArray(media)) {
+        return;
```

Comment:
Shouldn't we set the new map and return nothing?

Reply:
Good point! Yes, but it has no affect on the functionality and that is why I didn't notice it. Thanks for pointing it out.